model/task/db.go

```diff
@@ -985,72 +985,75 @@ func FindByExecutionTasksAndMaxExecution(taskIds []*string, execution int) ([]Ta
 
 type BuildVariantTuple struct {
 	BuildVariant string `bson:"build_variant"`
-	DisplayName  string `bson:"display_name"`
+	DisplayName  string `bson:"build_variant_display_name"`
 }
 
 const VersionLimit = 50
 
-// FindUniqueBuildVariantNamesByTask returns a list of unique build variants names and their display names for a given task name.
-// It attempts to return the most recent display name for each build variant to avoid returning duplicates caused by display names changing.
-// It only checks the last 50 versions that ran for a given task name.
-func FindUniqueBuildVariantNamesByTask(projectId string, taskName string, repoOrderNumber int) ([]*BuildVariantTuple, error) {
-	pipeline := []bson.M{
-		{"$match": bson.M{
-			ProjectKey:     projectId,
-			DisplayNameKey: taskName,
-			RequesterKey:   bson.M{"$in": evergreen.SystemVersionRequesterTypes},
-			"$and": []bson.M{
-				{RevisionOrderNumberKey: bson.M{"$gte": repoOrderNumber - VersionLimit}},
-				{RevisionOrderNumberKey: bson.M{"$lte": repoOrderNumber}},
-			},
+// Reusable pipeline stages when finding build variant display names
+var (
+	// sort build variant display names alphabetically
+	sortByVariantDisplayName = bson.M{
+		"$sort": bson.M{
+			"build_variant_display_name": 1,
```

Comment:
It would be better to make bsonutil keys for BuildVariantTuple's fields rather than hardcoding here.